### PR TITLE
Audit Logs Fixes #4

### LINKED ISSUE #4: Autdit Logs
Created PR #5

### PR DESCRIPTION
**Pull request by pi0**

### CODE REVIEW

**Review by Tabrizian — COMMENTED** [review]
There is a typo in the file name. autdit :)

**Comment by Tabrizian** [comment]
What's the usage of target and target model?

**Comment by pi0** [comment]
@Tabrizian see #4

File: lib/logging/audit/model.js [hunk 1]
```diff
@@ -35,6 +35,24 @@ class Audit extends Model {
         };
     }
 
+    static findBy(target, populate, action, limit) {
```
Comment (pi0) [inline lib/logging/audit/model.js]: Change params to object `findBy({target,populate,...})` this is preferred in libraries so API changes won't break project that use it

File: lib/logging/audit/model.js [hunk 2]
```diff
@@ -35,6 +35,24 @@ class Audit extends Model {
...
+        let audit;
+        if (action)
+            audit = this.find({target: target, action: action}).sort({created_at: -1});
```
Comment (pi0) [inline lib/logging/audit/model.js]: We need to extract _id from target if instance is provided! 
```js
let _target=undefined
if(target) {
 _target=target._id || target
}
```

File: lib/logging/audit/model.js [hunk 3]
```diff
@@ -35,6 +35,24 @@ class Audit extends Model {
...
+        else
+            audit = this.find({target: target}).sort({created_at: -1});
+
+        if (limit > 0) {
+            audit = audit.limit(limit);
+        }
+
+        if (populate) {
+            audit = audit.populate(populate, 'name roles');
```
Comment (pi0) [inline lib/logging/audit/model.js]: why we just hard coded only name and roles fields?

Comment (siavashkavousi) [reply lib/logging/audit/model.js]: It was from last impl :D

Comment (pi0) [reply lib/logging/audit/model.js]: maybe we can use `populate_fields` for second param

**Pull request merged**